### PR TITLE
Adding Sidecar CRD and renaming Sidecar role

### PR DESCRIPTION
**Pull request by rshriram**
This is prep for https://github.com/istio/istio/pull/10717

### CODE REVIEW

**Comment by rshriram** [comment]
force merging. Lint failure is unrelated and being tackled in other PRs

File: pilot/pkg/model/validation.go [hunk 1]
```diff
@@ -564,6 +570,171 @@ func ValidateEnvoyFilter(name, namespace string, msg proto.Message) (errs error)
...
+				portMap[i.Port.Number] = struct{}{}
+			}
+		}
+
+		// validate that the hosts field is a slash separated value
+		// of form ns1/host, or */host, or */*, or ns1/*, or ns1/*.example.com
+		if len(i.Hosts) == 0 {
+			errs = appendErrors(errs, fmt.Errorf("sidecar: egress listener must contain at least one host"))
+		} else {
+			for _, host := range i.Hosts {
```
Comment (costinm) [inline pilot/pkg/model/validation.go]: I think the spec was allowing importing names without requiring namespace.

Also t he second part should allow 'short name' if the namespace is already defined:

 "ns1/myscv.ns1"  seems redundant.

The common use will be to import things like 

   svc1.ns1
   svc2.ns2 
(i.e. other k8s/mesh services )

**Pull request merged**